frontend/src/k8sTypes.ts

```diff
@@ -589,6 +589,81 @@ export type DSPipelineKind = K8sResourceCommon & {
   };
 };
 
+export type ClusterQueueKind = K8sResourceCommon & {
+  spec: Partial<{
```

Comment:
I'll double check, but I was trying to do that here. Most of the fields in this CRD are optional.

Reply:
@andrewballentyne would it be better to avoid `Partial<>` here and instead explicitly have each optional field marked optional (even where all fields are optional)? I was just looking at what had been done for similar types

Reply:
Yes, I don't want anyone getting away with Partials on CRD types -- it's hard to validate and trust if it's all painted with one brush. However, I find it hard to believe the legitimacy of a CRD with no required attributes under spec -- seems like they missed adding the required field.

Perhaps this is a request for improvement on their side.

Reply:
This is the CRD I'm referencing: https://github.com/kubernetes-sigs/kueue/blob/main/config/components/crd/bases/kueue.x-k8s.io_clusterqueues.yaml

There are some required fields nested deeper, but nothing at the top of the spec unless I'm misreading it. I'll switch to the more explicit syntax though and check with DW folks.

Here's a pretty-printed view of all the kueue CRDs: https://doc.crds.dev/github.com/kubernetes-sigs/kueue